src/Microsoft.AspNetCore.Analyzer.Testing/TestSource.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.AspNetCore.Analyzer.Testing
+{
+    public class TestSource
+    {
+        private const string MarkerStart = "/*MM";
+        private const string MarkerEnd = "*/";
+
+        public IDictionary<string, DiagnosticLocation> MarkerLocations { get; } 
+            = new Dictionary<string, DiagnosticLocation>(StringComparer.Ordinal);
+
+        public DiagnosticLocation DefaultMarkerLocation { get; private set; }
+
+        public string Source { get; private set; }
+
+        public static TestSource Read(string rawSource)
```

Comment:
Moved this here. I think there's be value in also have the location assertion here (since it's a bunch of boiler plate code)